Firebase/Auth/Source/Public/FIRMultiFactor.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "FIRAuth.h"
+#import "FIRMultiFactorAssertion.h"
+#import "FIRMultiFactorInfo.h"
+#import "FIRMultiFactorSession.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef void (^FIRMultiFactorSessionCallback)(FIRMultiFactorSession *_Nullable, NSError *_Nullable);
+
+extern NSString *const _Nonnull FIRPhoneMultiFactorID NS_SWIFT_NAME(PhoneMultiFactorID);
+
+NS_SWIFT_NAME(MultiFactor)
+@interface FIRMultiFactor : NSObject
+
+@property(nonatomic, readonly) NSArray<FIRMultiFactorInfo *> *enrolledFactors;
+
+/** @fn getSessionWithCompletion:
+    @brief Get a session for a second factor enrollment operation.
+    @param completion A block with the session identifier for a second factor enrollment operation.
+        This is used to identify the current user trying to enroll a second factor.
+*/
+- (void)getSessionWithCompletion:(nullable FIRMultiFactorSessionCallback)completion;
+
+/** @fn enrollWithAssertion:displayName:completion:
```

Comment:
Since this only works on iOS, we should prevent it from being called from other platforms.

Reply:
Yeah, good catch. Platform specific macros added.